src/Core/Billing/Organizations/Commands/UpdateOrganizationLicenseCommand.cs

```diff
@@ -46,6 +47,46 @@ public async Task UpdateLicenseAsync(SelfHostedOrganizationDetails selfHostedOrg
         }
 
         var claimsPrincipal = _licensingService.GetClaimsPrincipalFromLicense(license);
+
+        // If the license has a Token (claims-based), extract all properties from claims BEFORE validation
+        // This ensures that CanUseLicense validation has access to the correct values from claims
+        // Otherwise, fall back to using the properties already on the license object (backward compatibility)
+        if (claimsPrincipal != null)
```

Comment:
I am thinking we should check for when the claimsPrincipal is null  and log a warning message or throw new BadRequestException("Invalid license token") ; is required.